File: APIGatewayService/src/main/java/dev/slethware/apigateway/controller/HealthController.java
package dev.slethware.apigateway.controller;

import dev.slethware.apigateway.dto.response.ApiResponse;
import dev.slethware.apigateway.dto.response.HealthResponse;
import dev.slethware.apigateway.service.HealthCheckService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/v1/health")
@RequiredArgsConstructor
public class HealthController {

    private final HealthCheckService healthCheckService;

    @GetMapping
    public ResponseEntity<ApiResponse<HealthResponse>> getHealth() {
        HealthResponse health = healthCheckService.checkHealth();
        return ResponseEntity.ok(ApiResponse.success(health, "Health check complete"));
    }
}

File: src/controllers/health.controller.ts
// src/controllers/health.controller.ts
import { Controller, Get } from '@nestjs/common';
import { HealthCheckService } from '../services/health-check.service';

@Controller('health')
export class HealthController {
  constructor(private readonly healthService: HealthCheckService) {}

  @Get()
  async checkHealth() {
    const health = await this.healthService.checkHealth();
    return {
      success: true,
      data: health,
      message: 'Health check completed',
    };
  }
}
